app_flutter/lib/service/appengine_cocoon.dart

```diff
@@ -26,22 +27,23 @@ class AppEngineCocoonService implements CocoonService {
   /// The Cocoon API endpoint to query
   ///
   /// This is the base for all API requests to cocoon
-  static const String _appengineAuthority = 'flutter-dashboard.appspot.com';
-  static const String _baseApiUrl = 'https://$_appengineAuthority/api';
+  static const String _baseApiUrl = 'https://flutter-dashboard.appspot.com';
 
   final http.Client _client;
 
   @override
   Future<CocoonResponse<List<CommitStatus>>> fetchCommitStatuses() async {
+    final String getStatusUrl = kIsWeb
+        ? '/api/public/get-status'
+        : '$_baseApiUrl/api/public/get-status';
+
```

Comment:
Consider encapsulate this pattern into a function, so that this pattern isn't scattered in 3 functions. For example,
```
// This function helps the Web app talks to a local Cocoon AppEngine instance when running locally, or talks to 
// the production instance when deployed into production.
// The urlSuffix begins with a slash, e.g. "/api/public/get-status".
String getEndpoint(String urlSuffix) {
  return kIsWeb ? urlSuffix : 'https://flutter-dashboard.appspot.com$urlSuffix;
}
```

Reply:
Thank you! Added.